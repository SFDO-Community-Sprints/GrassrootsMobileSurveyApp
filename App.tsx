import React, { useState, useEffect } from 'react';
import * as Font from 'expo-font';
import { StatusBar } from 'expo-status-bar';
import Toast from 'react-native-toast-message';

import Router from './src/Router';
import { initializeStorage } from './src/utility/storage';
import { LocalizationContextProvider } from './src/context/localizationContext';
import { AuthContextProvider } from './src/context/authContext';
import { SurveyEditorContextProvider } from './src/context/surveyEditorContext';
import { toastConfig } from './src/components/Toast';

export default function App() {
  const [fontLoaded, setFontLoaded] = useState(false);

  useEffect(() => {
    initializeStorage();
    const loadFont = async () => {
      await Font.loadAsync({
        'SalesforceSans-Regular': require('./assets/fonts/SalesforceSans-Regular.ttf'),
        'SalesforceSans-Bold': require('./assets/fonts/SalesforceSans-Bold.ttf'),
        'SalesforceSans-Light': require('./assets/fonts/SalesforceSans-Light.ttf'),
      });
      setFontLoaded(true);
    };
    loadFont();
  }, []);

  return fontLoaded ? (
    <LocalizationContextProvider>
      <AuthContextProvider>
        <SurveyEditorContextProvider>
          <Router />
        </SurveyEditorContextProvider>
      </AuthContextProvider>
      <StatusBar style="dark" />
      <Toast config={toastConfig} />
    </LocalizationContextProvider>
  ) : null;
}
