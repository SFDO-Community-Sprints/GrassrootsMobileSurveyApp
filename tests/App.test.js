import React from 'react';
import { render, waitFor } from '@testing-library/react-native';

import App from '../App';

jest.mock('expo-font');
jest.mock('expo-asset');
jest.mock('react-native-keyboard-aware-scroll-view', () => {
  const KeyboardAwareScrollView = require('react-native').ScrollView;
  return { KeyboardAwareScrollView };
});

describe('<App />', () => {
  it('Before font loading', async () => {
    const screen = render(<App />);
    await waitFor(() => {
      expect(screen.toJSON()).toMatchSnapshot();
    });
  });
});
