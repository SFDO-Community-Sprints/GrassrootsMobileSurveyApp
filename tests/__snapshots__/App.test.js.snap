// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Before font loading 1`] = `null`;
